### PR TITLE
Log routes

### PR DESCRIPTION
**Pull request by LoganWright**

### CODE REVIEW

File: Sources/Routing/Router+Routes.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+//  Routing
+//
+//  Created by Logan Wright on 11/1/16.
+//
+//
+
+import Foundation
+
+extension Router {
+    public func routes(includeHost: Bool = false) -> [String] {
```
Comment (LoganWright) [inline Sources/Routing/Router+Routes.swift]: Should we keep the host optional? And how would we prefer it formatted w/ host.

File: Sources/Routing/Router+Routes.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                base += "\(method) "
+                routes += branch.routes.map { base + $0 }
+            }
+        }
+        return routes
+    }
+}
+
+extension Branch {
+    public var routes: [String] {
```
Comment (LoganWright) [inline Sources/Routing/Router+Routes.swift]: maybe make internal as well unless people ask for it.

File: Tests/RoutingTests/RouteExtractionTests.swift [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+@testable import Routing
+import HTTP
+
+import Node
+
+enum Foo: String {
+    case a,b,c
+}
+
+extension Foo: NodeConvertible {
```
Comment (LoganWright) [inline Tests/RoutingTests/RouteExtractionTests.swift]: This is all a separate example and wasn't intended to be committed. RM

File: Tests/RoutingTests/RouteExtractionTests.swift [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        let handler: RequestHandler = { _ in return Response(body: "Rawr") }
+
+        let router = Router<RequestHandler>()
+        router.add(path: ["*", "GET", "foo", ":bar", "*"], value: handler)
+        router.add(path: ["*", "POST", "foo"], value: handler)
+        router.add(path: ["*", "PUT", "users", ":id"], value: handler)
+
+        let routes = router.routes(includeHost: true)
+        let expectation = [
+            "*: GET /foo/:bar",
```
Comment (LoganWright) [inline Tests/RoutingTests/RouteExtractionTests.swift]: I think dropping the colon would look better:

```
* GET /foo/:bar
0.0.0.0 PUT /match/route/:here/*
```

Comment (tannernelson) [inline Tests/RoutingTests/RouteExtractionTests.swift]: Agreed.

**Comment by tannernelson** [comment]
How would you print this from Vapor? Something like this?

```swift
print(drop.router)
```

**Comment by LoganWright** [comment]
@tanadeau you'd print like `print(drop.router.routes())`

**Comment by tannernelson** [comment]
Lol wrong "tan"

**Pull request merged**